.github/workflows/sonarcloud.yml

```
@@ -0,0 +1,56 @@
+name: sonar-cloud
+
+on:
+  push:
+    paths-ignore:
+      # specific folder locations
+      - ".vscode/**"
+      - "docs/**"
+      # filetypes
+      - "**.md"
+      - "**.rst"
+      - "**.ipynb"
+      - "**.cff"
+      - "**.png"
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore:
+      # specific folder locations
+      - ".vscode/**"
+      - "docs/**"
+      # filetypes
+      - "**.md"
+      - "**.rst"
+      - "**.ipynb"
+      - "**.cff"
+      - "**.png"
+jobs:
+  sonarcloud:
+    name: Tests coverage
```

like this the action is not run while the PR is still a draft

```suggestion
  sonarcloud:
    if: github.event.pull_request.draft == false
    name: Tests coverage
```